worlds/tunic/er_scripts.py

```diff
@@ -504,3 +504,28 @@ def update_reachable_regions(connected_regions: Set[str], traversal_reqs: Dict[s
         connected_regions = update_reachable_regions(connected_regions, traversal_reqs, has_laurels, logic)
 
     return connected_regions
+
+
+# sort the portal dict by the name of the first portal, referring to the portal order in the master portal list
+def sort_portals(portal_pairs: Dict[Portal, Portal]) -> Dict[str, str]:
+    sorted_pairs: Dict[str, str] = {}
+    reference_list: List[str] = [portal.name for portal in portal_mapping]
+    reference_list.append("Shop Portal")
+
+    # note: this is not necessary yet since the shop portals aren't numbered yet -- they will be when decoupled happens
+    # due to plando, there can be a variable number of shops
+    # I could either do it like this, or just go up to like 200, this seemed better
+    # shop_count = 0
+    # for portal1, portal2 in portal_pairs.items():
+    #     if portal1.name.startswith("Shop"):
+    #         shop_count += 1
+    #     if portal2.name.startswith("Shop"):
+    #         shop_count += 1
+    # reference_list.extend([f"Shop Portal {i + 1}" for i in range(shop_count)])
+
+    for name in reference_list:
+        for portal1, portal2 in portal_pairs.items():
+            if name == portal1.name:
+                sorted_pairs[portal1.name] = portal2.name
```

Comment:
actually, it would be nice if there could be a `break` statement after this line. (I think `break` only breaks out of the current for loop, might be wrong)

Reply:
Good call, didn't think about that!